lib/queue.rb

```diff
@@ -1,16 +1,35 @@
-class Queue
 
+class Queue
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+    if @front == @back
+      raise ArgumenetError
+    end
```

Comment:
I suggest making this an elsif otherwise front and back will have an extra space between them due to the fact that the 1st if sets back at 1.

```suggestion
    @back = 0
    elsif @front == @back
      raise ArgumentError, "Queue is full"
    end
```